docs/_docs/04_reference/cli-options.md

```diff
@@ -1522,3 +1523,25 @@ plan. Resource actions are indicated with the following symbols:
 
 OpenTofu will perform the following actions:
 ```
+
+### feature
+
+**CLI Arg**: `--feature`<br/>
+**Environment Variable**: `TG_FLAG`<br/>
+
+Feature flags in Terragrunt allow users to dynamically control configuration behavior through CLI arguments or environment variables.
+
+These flags enable a more flexible and controlled deployment process, particularly in monorepo contexts with interdependent infrastructure units.
+
+Setting a feature flag through CLI:
+
+```bash
+terragrunt --feature int_feature_flag=123 --feature bool_feature_flag=true apply
+```
+
+Setting feature flags through environment variables:
+
+```bash
+export TG_FLAG=int_feature_flag=123,bool_feature_flag=true
+terragrunt apply
+```
```

Comment:
I recommend showing some HCL here too so it's easy to see what these control.

Reply:
Added